src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2229,19 +2229,24 @@ void os::Linux::print_process_memory_info(outputStream* st) {
   // Print glibc outstanding allocations.
   // (note: there is no implementation of mallinfo for muslc)
 #ifdef __GLIBC__
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 33)
+  struct mallinfo2 mi = ::mallinfo2();
+  const size_t total_allocated = mi.uordblks;
+  const bool might_have_wrapped = false;
```

Comment:
Can you move declaration of might_have_wrapped - defaulted to false - outside the define? Same for total_allocated.